File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlParenthesizedExpression.cs
namespace T1.SqlSharp.Expressions;

public class SqlParenthesizedExpression : ISqlExpression
{
    public SqlType SqlType => SqlType.Group;
    public required ISqlExpression Inner { get; set; }
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_Group(this);
    }

    public string ToSql()
    {
        return $"({Inner.ToSql()})";
    }

}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlParenthesizedExpression.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlParenthesizedExpression implements ISqlExpression {
    SqlType: SqlType = SqlType.Group;
    Inner!: ISqlExpression;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_ParenthesizedExpression(this);
    }

    ToSql(): string {
        return `(${this.Inner.ToSql()})`;
    }
} 